src/app/components/home/home.component.ts

```diff
@@ -650,15 +650,15 @@ export class HomeComponent implements OnInit, AfterViewInit {
   handleFileWordClicked(filter: string, event?): void {
     this.showSidebar();
     if (event && event.shiftKey) { // Shift click to exclude
-      if (!this.settingsButtons['exclude'].toggled) {
-        this.settingsButtons['exclude'].toggled = true;
+      if (!this.settingsButtons['tagExclusion'].toggled) {
+        this.settingsButtons['tagExclusion'].toggled = true;
       }
-      this.onEnterKey(filter, 4); // 4th item is the `file` exlcude filter
+      this.onEnterKey(filter, 7); // 7th item is the `tag` exlcude filter
     } else {
-      if (!this.settingsButtons['file'].toggled) {
-        this.settingsButtons['file'].toggled = true;
+      if (!this.settingsButtons['tagIntersection'].toggled) {
```

Comment:
The `handleFileWordClicked` method gets called when a word in the file name in the top bar is clicked - so the old behavior (opening `file` search) should be maintained in that scenario: https://github.com/whyboris/Video-Hub-App/blob/master/src/app/components/home/home.component.html#L402

This is because I intended for users (and suspect the current ways users use the feature is) to click on a word in the file name and see other files that have that word.

Once _manual tags_ are added, and the tag search (_tagIntersection_) will be searching manual tags, the intended functionality will disappear.

Similarly the 'word cloud' currently is correlated with the file name not with the tags.

Perhaps we just make `handleFileWordClicked` include another parameter (say 'tags: bool' -- and depending on whether the parameter is true or false, it will search either file names or tags?

Also unlike before this change, the 'shift' key doesn't seem to affect when clicking on a word in the file name.

Reply:
Haha - I hadn't even realised that this was functionality! 😂 Are you sure 'shift' click worked before hand? I don't think the event gets passed through!

And I think we're slightly on different pages here - the idea of the `tag` filters are that they are customisable. As I discussed in #13, my idea is that you can have any combination of the following included in tags:
- clean file names
- clean folder names
- manual tags
- other file metadata (extensions etc.)

So since the top file name words are already cleaned, it makes sense to pop them in the tag filter if you have `clean file names` for tags enabled! Since they are the same, the functionality won't change.

Say you just enable manual tags for tagging, then we can have it pop top bar words into the file search instead - but I would prefer to do this once tagging is initially implemented! 👍 

Let me know if you don't get what I mean - want to be on the same page going forward! 😄

Reply:
Just in case you've missed it - what do you think of my thinking above? 😄

Reply:
Sorry it took so long to get back to your question / idea / proposal -- busy week.

I feel like I might be misunderstanding you. I'll try to summarize what I think you're recommending and you can correct whatever misunderstanding I have. 

Part of the communication problem may be that I hadn't coherently labeled what things are called. So in the text below I'll try to be more clear.

- *auto-tags* - the tags that are automatically extracted from the file names (auto-capturing two-word combinations like "Batman Returns" if you have many files that include that word combination).
- *manual-tags* - the feature not yet in _master_ but that I'm working on at the moment - allowing users to add additional string descriptions to any file.
- *filter* - a user-editable field in the sidebar that removes from view from the full list of videos items that don't fit some rule (e.g. don't contain the word "batman").

In #13 I think you're proposing that when the app generates the *auto-tags* list it _also_ include folder names on the auto-generated list (possibly colored differently; perhaps if the folder contains more than 5 videos inside it). If that's done, then clicking on any one of the entities should pop the folder name into a *filter* on the sidebar; but at the moment there isn't a place which would accept both a folder name and a file name - hence the need for some hybrid filter (other than the "magic" filter we already have).

Is this roughly correct?